protected/views/site/search.php

```diff
@@ -0,0 +1,68 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
+    <head>
+        <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+        <meta name="language" content="en" />
+        <title><?php echo $this->pageTitle; ?></title>
+    </head>
+    <body>
+	<form>
+<label for="Name">Name:</label><br>
+<input type="text" id="name" name="name" value= '<?php echo $name ?>'><br>
+<label for="Name">Style:</label><br>
+<input type="text" id="style" name="style" value= '<?php echo $style ?>'><br>
+<label for="Name">Hop:</label><br>
+<input type="text" id="hop" name="hop" value= '<?php echo $hop ?>'><br>
+<label for="Name">Yeast:</label><br>
+<input type="text" id="yeast" name="yeast" value= '<?php echo $yeast ?>'><br>
+<label for="Name">Malt:</label><br>
+<input type="text" id="malt" name="malt" value= '<?php echo $malt ?>'><br>
+<input type="submit" value="Submit">
+</form>
+
+
+        <table>
+        <thead>
+                <tr>
+                    <th>Beer</th>
+                    <th>Style</th>
+                    <th>Hops</th>
+                    <th>Yeasts</th>
+                    <th>Malts</th>
+                </tr>
+            </thead>
+            <tbody>
+                <?php foreach ($beers as $beer) { ?>
+                    <tr>
+                        <td><?php echo $beer['name']; ?></td>
+                        <td><?php if ($beer['style']) echo $beer['style']; ?></td>
+                        <td>
+                            <?php foreach ($beer['hops'] as $h) { ?>
+                                <?php echo $h; ?>,
+                            <?php } ?>
+                        </td>
+                        <td>
+                            <?php foreach ($beer['yeasts'] as $y) { ?>
+                                <?php echo $y; ?>,
+                            <?php } ?>
+                        </td>
+                        <td>
+                            <?php foreach ($beer['malts'] as $m) { ?>
+                                <?php echo $m; ?>,
+                            <?php } ?>
+                        </td>
+                    </tr>
+                <?php $lastBeerId = $beer['beerId']; } ?>
```

Comment:
LastBeerId is used with key-offset pagination.
Because results (regardless of search criteria) are ordered by BeerId collecting the next page's worth of results can be determined by simply searching for BeerId's higher than (offset by )the last BeerId (the key).

The primary advantage of key-offset pagination is that to requests the Nth page's worth of results you dont need to hit the data for the page 1 to n-1's worth of results and discard them.

There are some (in my mind, minor) disadvantages though:
A 'back' button for pagination just hits the browsers back button and doesnt allow a user to maintain the browsing history exactly.
You can only jump to the 'next' page, and not any arbitrary page number.
Given how human being use search engines I believe these are a minor price to pay for more efficient database access.